File: dotnet/CudaRS.Native/SdkModelSpec.cs
using System;
using System.Runtime.InteropServices;

namespace CudaRS.Native;

[StructLayout(LayoutKind.Sequential)]
public struct SdkModelSpec
{
    public IntPtr IdPtr;
    public nuint IdLen;
    public SdkModelKind Kind;
    public IntPtr ConfigJsonPtr;
    public nuint ConfigJsonLen;
}

File: cudars-ffi/src/sdk/sdk_model_spec.rs
use libc::c_char;

use cudars_core::ModelKind;

#[repr(C)]
pub struct SdkModelSpec {
    pub id_ptr: *const c_char,
    pub id_len: usize,
    pub kind: ModelKind,
    pub config_json_ptr: *const c_char,
    pub config_json_len: usize,
}
